cpp/util/Histogram.h

```diff
@@ -250,6 +250,81 @@ template<typename T> class Histogram
             m_local_histograms.local().increment(value_bin, weight);
         }
 
+        void reduceHistogramsInto(ManagedArray<T>& result)
```

Comment:
This could be turned into a template function. One template is `util::Histogram<T>*` and the other is `util::ManagedArray<T>*`. See `HistogramVector` vs `ThreadStorageVector`. Where should it go?